src/test/java/com/virtuslab/archunit/UIThreadUnsafeMethodInvocationsTestSuite.java

```diff
@@ -205,60 +190,27 @@ public void check(JavaMethod method, ConditionEvents events) {
 
   @Test
   public void only_ui_thread_unsafe_method_should_call_blocking_intellij_methods() {
-    methods()
+    noMethods()
         .that()
         .areNotAnnotatedWith(UIThreadUnsafe.class)
-        .and()
-        .areNotAnnotatedWith(IgnoreUIThreadUnsafeCalls.class)
-        .should(new ArchCondition<JavaMethod>("never call any blocking IntelliJ methods") {
-          @Override
-          public void check(JavaMethod method, ConditionEvents events) {
-            method.getCallsFromSelf().forEach(call -> {
-              val calledMethodFullName = call.getTarget().getFullName();
-              if (calledMethodFullName.equals("com.intellij.dvcs.push.PushController.push(boolean)")) {
-                String message = "a non-${UIThreadUnsafeName} method ${method.getFullName()} " +
-                    "calls blocking method ${calledMethodFullName}";
-                events.add(SimpleConditionEvent.violated(method, message));
-              }
-            });
-          }
-        })
+        .should(callAnyMethodsThat("are known to be blocking IntelliJ APIs",
+            (method, calledMethod) -> calledMethod.getFullName().equals("com.intellij.dvcs.push.PushController.push(boolean)")))
         .check(productionClasses);
   }
 
   @Test
   public void only_ui_thread_unsafe_method_should_call_git4idea_or_io_methods() {
```

Comment:
just to be clear...  the nature b/w this one and `only_ui_thread_unsafe_method_should_call_blocking_intellij_methods` is different, right? here we block all methods  and whitelist some methods while in the other place we block some methods only? hence, it would not make sense either to merge them or to split `only_ui_thread_unsafe_method_should_call_git4idea_or_io_methods`?

Reply:
Yeah, exactly... while `only_ui_thread_unsafe_method_should_call_blocking_intellij_methods` is for checking calls to UI thread **unsafe methods** from otherwise **safe packages**... `only_ui_thread_unsafe_method_should_call_git4idea_or_io_methods` is for for checking calls to UI thread **safe methods** from otherwise **unsafe packages**.